generator/src/main/java/org/javabubble/generator/site/ReadmeHTML.java

```diff
@@ -157,4 +168,17 @@ private DomContent copyright() {
 					text(".")));
 	}
 
+	private DomContent searchField() throws IOException {
+		try {
+			final var source = Files.readString(Paths.get(getClass().getResource(SEARCH_JS).toURI()), UTF_8);
```

Comment:
```suggestion
		final var source = new String(getClass().getResourceAsStream(SEARCH_JS).readAllBytes(), UTF_8);
```

This will not require conversation to URI --> try/catch block can be removed.

Reply:
I implemented your suggestion and removed the catch as you requested. But I kept the try as a try-with-resources to make sure the stream will be closed correctly.

Worth a discussion: There can still be a `NullPointerException`in case the JavaScript file containing the search code will not be found or is not readable/accessible. In that case with the new code there will just occur a `NullPointerException` on the stream object where in my old code it was catched and wrapped with a better description.

Reply:
Thanks!

The javascript file is not an external input. It is part of our static code. I wouldn't check static code at runtime. Otherwise you can basically add a null check before every variable that is de-referenced.